src/tests/multihost/alltests/test_inmemory_cache.py

```diff
@@ -0,0 +1,329 @@
+        with pytest.raises(AssertionError):
+            find_logs(multihost, "/tmp/stracefoo1", "/var/lib/sss/pipes/nss")
+        time.sleep(10)
+        elapsed_second = int(client.run_command('date "+%s"').stdout_text.split()[0]) - date_start
+        if elapsed_second < 35:
+            client.run_command("strace -o /tmp/stracefoo1 getent passwd foo1")
+            with pytest.raises(AssertionError):
+                find_logs(multihost,  "/tmp/stracefoo1", "/var/lib/sss/pipes/nss")
+        time.sleep(6)
+        client.run_command("strace -o /tmp/stracefoo1 getent passwd foo1")
+        find_logs(multihost, "/tmp/stracefoo1", "/var/lib/sss/pipes/nss")
+
+    @staticmethod
+    def test_time_out_300(multihost):
+        """
+        :title: Set default value of option Memcache timeout to 300
+        :id: a0b9ee92-96fd-11ed-bc4e-845cf3eff344
+        :steps:
+            1. Set default value of option Memcache timeout to 300
+            2. Validate the user using sss cache
+        :expectedresults:
+            1. Should succeed
+            2. Should succeed
+        """
+        client = multihost.client[0]
+        tools = sssdTools(multihost.client[0])
+        tools.sssd_conf("nss", {'memcache_timeout': ''}, action='delete')
+        tools.clear_sssd_cache()
+        clear_only_domain_log(multihost)
+        client.run_command("strace -o /tmp/stracefoo1 getent passwd foo1")
+        find_logs(multihost, "/tmp/stracefoo1", "/var/lib/sss/pipes/nss")
+        time.sleep(100)
+        client.run_command("strace -o /tmp/stracefoo1 getent passwd foo1")
+        with pytest.raises(AssertionError):
+            find_logs(multihost, "/tmp/stracefoo1", "/var/lib/sss/pipes/nss")
+        time.sleep(100)
+        client.run_command("strace -o /tmp/stracefoo1 getent passwd foo1")
+        with pytest.raises(AssertionError):
+            find_logs(multihost, "/tmp/stracefoo1", "/var/lib/sss/pipes/nss")
+        time.sleep(101)
+        client.run_command("strace -o /tmp/stracefoo1 getent passwd foo1")
+        find_logs(multihost, "/tmp/stracefoo1", "/var/lib/sss/pipes/nss")
+
+    @staticmethod
+    def test_default_behaviour(multihost):
+        """
+        :title: Verify default behaviour of Cache Initgroups feature
+        :id: 975d9060-96fd-11ed-b9f7-845cf3eff344
+        :steps:
+            1. Set Memcache timeout default
+            2. Set Memcache timeout 0 and expire cache immediately
+            3. Invalidate the user using sss cache
+            4. Invalidate the group using sss cache
+            5. Stop sssd service and verify the cache
+        :expectedresults:
+            1. Should succeed
+            2. Should succeed
+            3. Should succeed
+            4. Should succeed
+            5. Should succeed
+        """
+        client = multihost.client[0]
+        tools = sssdTools(multihost.client[0])
+        tools.sssd_conf("nss", {'memcache_timeout': ''}, action='delete')
```

Comment:
You are changing the config, backupsssdconf fixture should be used to restore the config.